src/server.py

```diff
@@ -27,38 +31,53 @@ def server():
             if len(part) < buffer_length:
                 message_complete = True
         print(message)
-        conn.send(response_ok())
+        try:
+            parse_request(message)
+        except HTTPException as exception:
+            conn.send((response_error(exception).encode("utf-8")))
+            conn.close()
+        else:
+            conn.send(response_ok())
```

Comment:
See how you're calling conn.send in a couple of places?  This is not so great.  could you  instead bind a symbol to the response created and only call `response_ok` once?